test/commands.test.ts

```diff
@@ -0,0 +1,259 @@
+import * as vscode from 'vscode';
+import * as assert from 'assert';
+
+interface TestEditorOptions {
+    language?: string;
+    content?: string;
+}
+
+type TestEditorAction = (editor: vscode.TextEditor, document: vscode.TextDocument) => void;
+
+async function inTextEditor(options: TestEditorOptions, action: TestEditorAction) {
+    const d = await vscode.workspace.openTextDocument(options);
+    await vscode.window.showTextDocument(d);
+    await action(vscode.window.activeTextEditor!, d);
+}
+
+function move(editor: vscode.TextEditor, line: number, col: number) {
+    const pos = new vscode.Position(line, col);
+    editor.selection = new vscode.Selection(pos, pos);
+}
+
+function select(editor: vscode.TextEditor, range: vscode.Range) {
+    editor.selection = new vscode.Selection(range.start, range.end);
+}
+
+const weekdayArray = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+
+suite('Commands', () => {
+
+    test('Demote', async () => {
+        const steps = [
+            '* Home',
+            '** Home',
+            '*** Home',
+            '**** Home',
+            '***** Home',
+        ];
+
+        await inTextEditor({ language: 'org', content: steps[0] }, async (e, d) => {
```

Comment:
I would have prefer the use of `editor` and `document`for better clarity

Reply:
Renamed everywhere. Tests where `editor` is not used, I renamed it to `_`.